### PR TITLE
MIDI Testing

### PR DESCRIPTION
**Pull request by ed-xmos**
https://xmosjira.atlassian.net/browse/UA-21

### CODE REVIEW

File: tests/test_midi_stress.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    wait_for_midi_ports,
+    find_xmos_midi_device,
+    XrunDut,
+    XsigInput,
+    XsigOutput,
+    xsig_completion_time_s
+)
+from conftest import list_configs, get_config_features
+
+def midi_common_uncollect(features, board, pytestconfig):
```
Comment (danielpieczko) [inline tests/test_midi_stress.py:29]: Same here.

File: tests/test_midi_stress.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        # Ensure firmware is up and enumerated as MIDI
+        wait_for_midi_ports()
+
+        time_start = time.time()
+        with (
+            AudioAnalyzerHarness(
+                adapter_harness, xscope="io"
+            ) as harness,
+            # Due to in and out in xsig config this streams audio in both directions for max stress
+            XsigInput(fs_audio, duration, xsig_config_path, dut.dev_name, ident=f"analogue_input-{board}-{config}-{fs_audio}") as xsig_proc_in
```
Comment (danielpieczko) [inline tests/test_midi_stress.py:91]: Replace "analogue_input" in the `ident` parameter to uniquely identify this test.

File: tests/test_midi_stress.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            fail_str += f"Failure at sample rate {fs_audio}\n"
+            fail_str += "\n".join(failures) + "\n\n"
+            fail_str += f"xscope stdout at sample rate {fs_audio}\n"
+            fail_str += "\n".join(xscope_lines) + "\n\n"
+            fail_str += f"xsig stdout at sample rate {fs_audio}\n"
+            fail_str += "\n".join(xsig_lines) + "\n\n"
+
+    if len(fail_str) > 0:
+        print(fail_str)
+        # pytest.fail(fail_str)
```
Comment (danielpieczko) [inline tests/test_midi_stress.py:121]: Remove print and reinstate the pytest.fail()

**Review by danielpieczko — APPROVED** [review]
Tests look good, just some small points that need addressing.

File: Jenkinsfile [hunk 4]
```diff
@@ -201,7 +201,11 @@ pipeline {
 
                       withXTAG(["usb_audio_mc_xs2_dut", "usb_audio_mc_xs2_harness", \
                                 "usb_audio_xcai_exp_dut", "usb_audio_xcai_exp_harness"]) { xtagIds ->
-                        sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_mac_intel.xml \
+                        // sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_mac_intel.xml \
+                        //     -o xk_216_mc_dut=${xtagIds[0]} -o xk_216_mc_harness=${xtagIds[1]} \
+                        //     -o xk_evk_xu316_dut=${xtagIds[2]} -o xk_evk_xu316_harness=${xtagIds[3]}"
+                        // Only test MIDI for now
+                        sh "pytest test_midi.py test_midi_stress.py -s -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_mac_intel.xml \
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply Jenkinsfile:208]: Oops - thought I caught all of those.

File: tests/test_midi_stress.py (hunk 3 above)
Comment (ed-xmos) [reply tests/test_midi_stress.py:121]: Good catch

**Review by danielpieczko — APPROVED** [review]

File: Jenkinsfile [hunk 5]
```diff
@@ -385,7 +385,7 @@ pipeline {
                     }
 
                     withXTAG(["usb_audio_mc_xcai_dut", "usb_audio_mc_xcai_harness"]) { xtagIds ->
-                      sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_windows11.xml \
+                      sh "pytest -v --level ${params.TEST_LEVEL} --junitxml=pytest_result_windows10.xml \
```
Comment (danielpieczko) [inline Jenkinsfile:388]: ?

**Review by danielpieczko — APPROVED** [review]

File: Jenkinsfile (hunk 5 above)
Comment (ed-xmos) [reply Jenkinsfile:388]: Ah, some fine pasta there. Fixed..

**Review by xross — APPROVED** [review]

**Pull request merged**